lib/yabeda/datadog/units.rb

```diff
@@ -4,31 +4,44 @@ module Yabeda
-                   "device", "update", "method", "job", "container", "execution",
-                   "throttle", "invocation", "user", "success", "build", "prediction",
-                   "table", "index", "lock", "transaction", "query", "row", "key",
-                   "command", "offset", "record", "object", "cursor", "assertion", "scan",
-                   "document", "shard", "flush", "merge", "refresh", "fetch", "column",
-                   "commit", "wait", "ticket", "question", "hit", "miss", "eviction",
-                   "get", "set", "dollar", "cent", "page", "split", "hertz", "kilohertz",
-                   "megahertz", "gigahertz", "entry", "degree celsius", "degree fahrenheit",
-                   "nanocore", "microcore", "millicore", "core", "kilocore", "megacore",
-                   "gigacore", "teracore", "petacore", "exacore",].freeze
+      AVAILABLE = [
+        "bit", "byte", "kibibyte", "mebibyte", "gibibyte", "tebibyte",
+        "pebibyte", "exbibyte", "nanosecond", "microsecond", "millisecond",
+        "second", "minute", "hour", "day", "week", "percent_nano", "percent",
+        "apdex", "fraction", "connection", "request", "packet", "segment",
+        "response", "message", "payload", "timeout", "datagram", "route",
+        "session", "process", "thread", "host", "node", "fault", "service",
+        "instance", "cpu", "file", "inode", "sector", "block", "buffer",
+        "error", "read", "write", "occurrence", "event", "time", "unit",
+        "operation", "item", "task", "worker", "resource", "garbage collection",
+        "email", "sample", "stage", "monitor", "location", "check", "attempt",
+        "device", "update", "method", "job", "container", "execution",
+        "throttle", "invocation", "user", "success", "build", "prediction",
+        "table", "index", "lock", "transaction", "query", "row", "key",
+        "command", "offset", "record", "object", "cursor", "assertion", "scan",
+        "document", "shard", "flush", "merge", "refresh", "fetch", "column",
+        "commit", "wait", "ticket", "question", "hit", "miss", "eviction",
+        "get", "set", "dollar", "cent", "page", "split", "hertz", "kilohertz",
+        "megahertz", "gigahertz", "entry", "degree celsius", "degree fahrenheit",
+        "nanocore", "microcore", "millicore", "core", "kilocore", "megacore",
+        "gigacore", "teracore", "petacore", "exacore",
+      ].map { |unit| [unit, unit] }.to_h.freeze
 
-      # Find valid unit
+      IRREGULAR_PLURALS = {
+        "queries" => "query",
+        "entries" => "entry",
+        "indices" => "index",
+      }.freeze
+
+      # Find an available unit of metric
       def self.find(unit)
-        unit if unit && AVAILABLE.include?(unit)
+        unit_key = unit.to_s.downcase
+        AVAILABLE[unit_key] ||
+          AVAILABLE[unit_key.gsub(/s\z/, "")] ||
+          AVAILABLE[unit_key.gsub(/es\z/, "")] ||
+          IRREGULAR_PLURALS[unit_key]
```

Comment:
Reverse order of checks looks more natural for me: 0. look for exact match 1. look for irregular things, 2. try to strip suffixes from longest to shortest.

It was a bit unclear for me to understand why it works as expected here :-D

What do you think?

Reply:
> Reverse order of checks looks more natural for me: 0. look for exact match 1. look for irregular things, 2. try to strip suffixes from longest to shortest.

I thought that exact check should be first, I expect that the adapter users will use Datadog singular names for units, then names in plural more common form with 's' suffix, then less common plurals with 'es' suffix and the last rare irregular plurals. Does it make sense?

Perhaps the naming of constants misled you?

> It was a bit unclear for me to understand why it works as expected here :-D

I can not understand what you mean

Reply:
Make sense.

> > It was a bit unclear for me to understand why it works as expected here :-D
>
> I can not understand what you mean

At first glance, I thought that it should give wrong results, but it works correctly because we always check for unit presence in `AVAILABLE` hash. If this check was removed things definitely would break. It required a bit of thinking effort from me to understand.